src/org/jgroups/protocols/dns/DefaultDNSResolver.java

```diff
@@ -114,4 +115,17 @@ protected DirContext getDnsContext() {
         }
         return addresses;
     }
+
+    protected List<Address> resolveAEntries(String dnsQuery, String srcPort) {
```

Comment:
I believe this logic should be embedded into the existing `resolveAEntries` method. Perhaps we should try to parse the port. If successful, then we use it, if not, than we can use `0`.

Reply:
I am not sure what you mean. We can't add `srcPort` to the existing method for backwards compatibility. Do you mean that we should add it to the dnsQuery parameter? 
EG: `dnsQuery = dnsQuery:srcPort`
Then strip it out of the value if there and default to 0 if not there?

Reply:
This method used to be private, so I don't think there would be any issues regarding the interface change.  If it's really a concern, you could consider making the two Resolver classes package scope, to make sure they're not used outside DNS_PING.

Reply:
@slaskawi @rcernich Ok. I pushed a change based on this feedback.

Reply:
Great thanks!